catalog/spatial/registry/registry-schema-bindings/src/main/java/org/codice/ddf/registry/schemabindings/RegistryPackageWebConverter.java

```diff
@@ -0,0 +1,1760 @@
+                    EXTERNAL_IDENTIFIER_KEY), registryObject.getExternalIdentifier());
+            if (!registryObject.getExternalIdentifier()
+                    .isEmpty()) {
+                populated = true;
+            }
+        }
+
+        valueToPopulate = getStringFromMap(HOME_KEY, map);
+        if (StringUtils.isNotBlank(valueToPopulate)) {
+            registryObject.setHome(valueToPopulate);
+            populated = true;
+        }
+
+        valueToPopulate = getStringFromMap(ID_KEY, map);
+        if (StringUtils.isNotBlank(valueToPopulate)) {
+            registryObject.setId(valueToPopulate);
+            populated = true;
+        }
+
+        valueToPopulate = getStringFromMap(LID_KEY, map);
+        if (StringUtils.isNotBlank(valueToPopulate)) {
+            registryObject.setLid(valueToPopulate);
+            populated = true;
+        }
+
+        valueToPopulate = getStringFromMap(NAME_KEY, map);
+        if (StringUtils.isNotBlank(valueToPopulate)) {
+            InternationalStringType istToPopulate =
+                    RegistryPackageUtils.getInternationalStringTypeFromString(valueToPopulate);
+            if (istToPopulate != null) {
+                registryObject.setName(istToPopulate);
+                populated = true;
+            }
+        }
+
+        valueToPopulate = getStringFromMap(OBJECT_TYPE_KEY, map);
+        if (StringUtils.isNotBlank(valueToPopulate)) {
+            registryObject.setObjectType(valueToPopulate);
+            populated = true;
+        }
+
+        if (map.containsKey(SLOT)) {
+            populateSlotList((List<Map<String, Object>>) map.get(SLOT), registryObject.getSlot());
+            if (!registryObject.getSlot()
+                    .isEmpty()) {
+                populated = true;
+            }
+        }
+
+        valueToPopulate = getStringFromMap(STATUS_KEY, map);
+        if (StringUtils.isNotBlank(valueToPopulate)) {
+            registryObject.setStatus(valueToPopulate);
+            populated = true;
+        }
+
+        valueToPopulate = getStringFromMap(VERSION_INFO_KEY, map);
+        if (StringUtils.isNotBlank(valueToPopulate)) {
+            VersionInfoType versionInfo = RIM_FACTORY.createVersionInfoType();
+            versionInfo.setVersionName(valueToPopulate);
+
+            registryObject.setVersionInfo(versionInfo);
+            populated = true;
+        }
+
+        return populated;
+    }
+
+    private static String getStringFromMap(String key, Map<String, Object> map) {
+        String value = null;
+        if (map.containsKey(key)) {
+            value = (String) map.get(key);
+        }
+
+        return value;
+    }
+
+    private static BigInteger getBigIntFromMap(String key, Map<String, Object> map) {
+        BigInteger value = null;
+        if (map.containsKey(key)) {
+            // Using String.valueOf so that it can handle Integer and Long the same
+            value = new BigInteger(String.valueOf(map.get(key)));
```

Comment:
What happens here if `BigInteger` throws an exception because the string isn't number?

Also, I'm not sure why we're using `String.valueOf()` here and why we have that comment. Couldn't we just call our `getStringFromMap()` and have something like this (with exception handling if needed)?

```
return new BigInteger(getStringFromMap(key, map));
```